lib/blockchain.go

```diff
@@ -517,9 +538,98 @@ type Blockchain struct {
 	syncingState                bool
 	downloadingHistoricalBlocks bool
 
+	// checkpointSyncingProviders is a list of providers from which we will request the committed tip block info
+	// when syncing. The committed tip block info is used to designate a checkpoint before which signature
+	// verification will be skipped. These checkpoint providers should be trusted and should be able to provide
+	// the committed tip block info for the chain we are syncing.
+	checkpointSyncingProviders []string
+	// checkpointBlockInfo is the latest checkpoint block info that we have received from the checkpoint syncing
+	// providers.
+	checkpointBlockInfo *CheckpointBlockInfo
+
 	timer *Timer
 }
 
+func (bc *Blockchain) updateCheckpointBlockInfo() {
+	if len(bc.checkpointSyncingProviders) == 0 {
+		glog.V(2).Info("updateCheckpointBlockInfo: No checkpoint syncing providers set. Skipping update.")
+		return
+	}
+	var wg sync.WaitGroup
```

Comment:
Nit: I don't *think* you need the waitgroup here because the for-loop where you consume from the channel using `<-ch` will naturally force blocking until they all return.